pkg/products/marin3r/apiUsagePrometheusRules.go

```diff
@@ -0,0 +1,137 @@
+package marin3r
+
+import (
+	"errors"
+	"fmt"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var (
+	level1ApiUsageLowerThresholdPercent  = 80
+	level1ApiUsageHigherThresholdPercent = 90
+	level1ApiUsageCheckFrequencyMins     = 320
+	level2ApiUsageLowerThresholdPercent  = 90
+	level2ApiUsageHigherThresholdPercent = 95
+	level2ApiUsageCheckFrequencyMins     = 120
+	level3ApiUsageLowerThresholdPercent  = 95
+	level3ApiUsageCheckFrequencyMins     = 30
+)
+
+func (r *Reconciler) newAlertsReconciler(rateLimitUnit string, rateLimitRequestsPerUnit uint32) (resources.AlertReconciler, error) {
+
+	requestsAllowedPerSecond, err := getRateLimitInSeconds(rateLimitUnit, rateLimitRequestsPerUnit)
+	if err != nil {
+		return nil, err
+	}
+
+	level1Rule, err := getLevel1ApiUsageAlert(rateLimitUnit, rateLimitRequestsPerUnit, requestsAllowedPerSecond)
+	if err != nil {
+		return nil, err
+	}
+	level2Rule, err := getLevel2ApiUsageAlert(rateLimitUnit, rateLimitRequestsPerUnit, requestsAllowedPerSecond)
+	if err != nil {
+		return nil, err
+	}
+	level3Rule, err := getLevel3ApiUsageAlert(rateLimitUnit, rateLimitRequestsPerUnit, requestsAllowedPerSecond)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resources.AlertReconcilerImpl{
+		ProductName:  "3Scale",
+		Installation: r.installation,
+		Logger:       r.logger,
+		Alerts: []resources.AlertConfiguration{
+			{
+				AlertName: "3scale-api-usage-alert-level1",
```

Comment:
Can probably remove 3scale from the alert name and group. As these will be visible to the customer I'm not sure we want to reference 3scale directly given that the product name is Openshift API Management. `api-usage-alert-level1` is probably good for now